extra_geom/detectors.py

```diff
@@ -993,6 +999,151 @@ def to_distortion_array(self, allow_negative_xy=False):
         return super().to_distortion_array(allow_negative_xy)
 
 
+class LPD_MiniGeometry(LPDGeometryBase):
+    """Detector Layout for LPD Mini.
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    You won't normally instantiate this class directly:
+    use one of the constructor class methods to create or load a geometry.
+    """
+    detector_type_name = 'LPD-Mini'
+    pixel_size = 5e-4  # 5e-4 metres == 0.5 mm
+    n_tiles_per_module = 2
+    expected_data_shape = (0, 32, 256)
+    _pyfai_cls_name = 'LPDMINI'
+
+    def __init__(self, modules, filename='No file', metadata=None):
+        super().__init__(modules, filename, metadata)
+        self.expected_data_shape = (len(modules), 32, 256)
+        self.n_modules = len(modules)
+
+    @classmethod
+    def from_module_positions(cls, positions, rotations=None, asic_gap=None, unit=1e-3):
```

Comment:
Now that I've written the support for in-plane rotations (which did involve some swearing getting all the signs right)... is it actually necessary? We could lay the modules out without rotation, and then use `.rotate()` afterwards.

I was prompted by a diagram showing that a planned use case has two modules turned on their side and stacked together vertically (making a column of 64 px width by 256 pixels height).

I'm inclined to leave it in, because to define the geometry with the `rotate()` method, you'd either have to figure out fake corner positions so that they rotate into the right places, or fiddle with specifying the corner position of each module as the centre for its rotation.

We could also come up with some way of specifying it that only allows rotations in multiples of 90 degrees, to simplify things a bit, but now that I've (hopefully) got it working, I don't think there's much advantage to that.